worker/raft/worker.go

```diff
@@ -292,7 +346,22 @@ func (w *Worker) loop(raftConfig *raft.Config) (loopErr error) {
 			// the local node was removed from the cluster
 			// configuration, causing it to shutdown.
 			return errors.New("raft shutdown")
+		case <-noLeaderCheck:
+			noLeaderCheck = w.config.Clock.After(noLeaderFrequency)
+			if r.State() == raft.Leader {
+				lastContact = w.config.Clock.Now()
+				continue
+			}
+			var zeroTime time.Time
+			if r.LastContact() != zeroTime {
```

Comment:
if latest := r.LastContact(); latest != zeroTime {
    lastContact = latest
}

Reply:
Yeah, that's nicer - thanks!